docs/source/guide/s3.rst

```diff
@@ -0,0 +1,112 @@
+        Params={
+            'Bucket': 'bucket-name',
+            'Key': 'key-name'
+        }
+    )
+
+    # Use the URL to perform the GET operation. You can use any method you like
+    # to send the GET, but we will use requests here to keep things simple.
+    response = requests.get(url)
+
+Generating Presigned POSTs
+--------------------------
+
+Much like pre-signed URLs, pre-signed POSTs allow you to give write access to a
+user without giving them AWS credentials. The information you need to make the
+POST is returned by the :py:meth: `S3.Client.generate_presigned_post` method::
+
+    import boto3
+    import requests
+
+    # Get the service client
+    s3 = boto3.client('s3')
+
+    # Generate the POST attributes
+    post = s3.generate_presigned_post(
+        Bucket='bucket-name',
+        Key='key-name'
+    )
+
+    # Use the returned values to POST an object. Note that you need to use ALL
+    # of the returned fields in your post. You can use any method you like to
+    # send the POST, but we will use requests here to keep things simple.
+    files = {"file": "file_content"}
+    response = requests.post(post["url"], data=post["fields"], files=files)
+
+When generating these POSTs, you may wish to auto fill certain fields or
+constrain what your users submit. You can do this by providing those fields and
+conditions when you generate the POST data.::
+
+    import boto3
+
+    # Get the service client
+    s3 = boto3.client('s3')
+
+    # Make sure everything posted is publicly readable
+    fields = {"acl": "public-read"}
+
+    # Ensure that the ACL isn't changed and restrict the user to a length
+    # between 10 and 100.
+    conditions = [
+        {"acl": "public-read"},
+        ["content-length-range", 10, 100]
+    ]
+
+    # Generate the POST attributes
+    post = s3.generage_presigned_post(
+        Bucket='bucket-name',
+        Key='key-name'
+    )
+
+If your bucket requires the use of signature version 4, you can elect to use it
+to sign your POST. This does not fundamentally change how you use generator,
+you only need to make sure that the client used has signature version 4
+configured.::
+
+    import boto3
+    import requests
+    from botocore.client import Config
+
+    # Get the service client with sigv4 configured
+    s3 = boto3.client('s3', config=Config(signature_version='s3v4'))
+
+    # Generate the POST attributes. There are no special fields or conidtions
+    # you need to pass in to make sigv4 work.
+    post = s3.generate_presigned_post(
+        Bucket='bucket-name',
+        Key='key-name'
+    )
+
+    # Use the returned values to POST an object. Note that you need to use ALL
+    # of the returned fields in your post.
+    files = {"file": "file_content"}
+    response = requests.post(post["url"], data=post["fields"], files=files)
+
+.. _transfer manager reference: http://boto3.readthedocs.org/en/latest/reference/customizations/s3.html#module-boto3.s3.transfer
```

Comment:
Can we at least add a small snippet about how to use upload/download file?

One of the issues I keep seeing is that be linking to these docs directly, people are manually creating `S3Transfer` objects, unaware of the fact that these are available directly on a client.

If you don't have bandwidth, I can add this, but I feel like uploading/downloading a file makes sense in the s3 user guide, and it is something we customize in the SDK.